### PR TITLE
Add package for approvaltests

### PR DESCRIPTION
**Pull request by dheater**
Specify library name and version:  **approvaltests.cpp/8.0.0**

- [ x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [ x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by dheater** [comment]
Closes #543 
See also the previous recipe for ApprovalTests.cpp version 7.0.0 #544 

Note: This version does not support [Boost].UT as there is no existing Conan recipe for that test framework. (I mistakenly thought that it was the same as Boost.test which it is not)

cc @claremacrae, @uilianries,  @p-podsiadly

**Comment by claremacrae** [comment]
Thanks @dheater, that was quick!

I don't know enough about Conan to be confident of this question, but should I investigate the 3 patch files? I was hoping that no patching would be needed now...

**Comment by dheater** [comment]
> Thanks @dheater, that was quick!
> 
> I don't know enough about Conan to be confident of this question, but should I investigate the 3 patch files? I was hoping that no patching would be needed now...

Thanks @claremacrae . You are correct. The patches are no longer needed. I just forgot to delete them when copying the recipe from 7.0.0. They're gone now.

File: recipes/approvaltests.cpp/8.0.0/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+CONAN_BASIC_SETUP()
+
+ADD_EXECUTABLE(test_package test_catch.cpp)
+#ADD_EXECUTABLE(test_package test_gtest.cpp)
```
Comment (SSE4) [inline recipes/approvaltests.cpp/8.0.0/test_package/CMakeLists.txt]: pls remove commented code
(or alternatively, compile executable according to the option activated)

Comment (dheater) [reply recipes/approvaltests.cpp/8.0.0/test_package/CMakeLists.txt]: @SSE4  I have those in as a quick way to test against the alternate unit test frameworks. The problems is each test_framework provides its own `main`. Is there a way to build the test package to test all three? Maybe multiple directories/executables?

Comment (SSE4) [reply recipes/approvaltests.cpp/8.0.0/test_package/CMakeLists.txt]: take a look at boost recipe:
https://github.com/conan-io/conan-center-index/blob/master/recipes/boost/all/test_package/conanfile.py#L17
https://github.com/conan-io/conan-center-index/blob/master/recipes/boost/all/test_package/CMakeLists.txt#L34
the idea is that you pass options to the CMake from the test package conanfile.
then, in CMake, you may control which executable(s) to build (or which sources to use)

**Comment by dheater** [comment]
I managed to royally screw up this branch so resubmitting as PR #741

**Pull request closed without merging**